File: tools/calibrate-pipeline/fixtures/ast-diff-trivial-check/good-04-java-real-migration/input.java
package com.acme.shop.tests;

import org.junit.jupiter.api.AfterEach;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;

import static org.junit.jupiter.api.Assertions.assertTrue;

public class LoginTest {

    private WebDriver driver;
    private WebDriverWait wait;

    @BeforeEach
    void setUp() {
        driver = new ChromeDriver();
        wait = new WebDriverWait(driver, Duration.ofSeconds(10));
        driver.get("https://app.acme.test/login");
    }

    @AfterEach
    void tearDown() {
        driver.quit();
    }

    @Test
    void validCredentialsLandOnDashboard() throws InterruptedException {
        driver.findElement(By.id("email")).sendKeys("jane@acme.test");
        driver.findElement(By.id("password")).sendKeys("hunter2");
        driver.findElement(By.cssSelector("button.primary")).click();
        Thread.sleep(2000);
        WebElement dashboard = wait.until(
            ExpectedConditions.visibilityOfElementLocated(By.cssSelector(".dashboard-greeting"))
        );
        assertTrue(dashboard.isDisplayed());
    }
}

File: tools/calibrate-pipeline/fixtures/ast-diff-trivial-check/good-05-py-real-migration/input.py
import time

import pytest
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support import expected_conditions as EC
from selenium.webdriver.support.ui import WebDriverWait


@pytest.fixture
def driver():
    drv = webdriver.Chrome()
    drv.implicitly_wait(5)
    yield drv
    drv.quit()


def test_user_can_log_in(driver):
    driver.get("https://hr.beacon.test/login")
    driver.find_element(By.ID, "email").send_keys("hr-admin@beacon.test")
    driver.find_element(By.ID, "password").send_keys("Sup3rSecret!")
    driver.find_element(By.XPATH, "//form//button[@type='submit']").click()

    time.sleep(2)

    WebDriverWait(driver, 10).until(
        EC.visibility_of_element_located((By.CSS_SELECTOR, ".dashboard-greeting"))
    )

    greeting = driver.find_element(By.CSS_SELECTOR, ".dashboard-greeting").text
    assert greeting == "Welcome back, HR Admin"
